### PR TITLE
Add first replay test

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@Hixie

**Comment by tvolkert** [comment]
FYI, the full recording that this checks in is 350 KB

File: dev/bots/test.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import 'package:path/path.dart' as p;
 String flutterRoot = p.dirname(p.dirname(p.dirname(p.fromUri(Platform.script))));
 String flutter = p.join(flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter');
 String dart = p.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'dart.exe' : 'dart');
+String pub = p.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'pub.exe' : 'pub');
```
Comment (goderbauer) [inline dev/bots/test.dart]: On Windows, the executable is currently called `pub.bat`.

**Comment by Hixie** [comment]
LGTM, but travis is angry.

**Comment by Hixie** [comment]
We talked about future plans. The idea that made the most sense to us is having tests first run their test against a boring environment (we would have one checked in for each platform we support), then tweak the (in-memory) recording to be what we're actually testing for (e.g. "pub segfaulted"), and then run the test against that mutated recording and see if the tool reacts as we would like.

**Comment by tvolkert** [comment]
Ok, all automated tests are passing now.  This is probably worth another look.  /cc @yjbanov who was recently in this code as well.

**Comment by yjbanov** [comment]
A bit late to the party, but can we store a recording in one file?

**Comment by tvolkert** [comment]
A recording gets zipped up into one file when it takes the form of a bug report, but one file for all that's in a recording would be too unwieldy.

It's kinda moot with respect to our replay tests, since this was just a proof-of-concept that also ensures that we're able to run hermetically and don't regress while I work on the longer-term solution, which won't require us checking in recordings at all. (see Ian's comment above)

**Comment by yjbanov** [comment]
Great plan!

**Pull request merged**